### ISSUE TITLE
False negative "node/no-extraneous-require" behavior

### ISSUE THREAD

**Issue by ro0gr** [mysticatea/eslint-plugin-node#107]
This issue reproduces on `ember-cli`'s CI:

https://travis-ci.org/ember-cli/ember-cli/jobs/335394186#L581-L607:
>>
```
 lib/tasks/server/middleware/broccoli-serve-files/index.js
     3:24  error  "heimdalljs-logger" is extraneous    node/no-extraneous-require
    22:40  error  "broccoli-middleware" is extraneous  node/no-extraneous-require
  
  lib/tasks/server/middleware/broccoli-watcher/index.js
     3:30  error  "clean-base-url" is extraneous       node/no-extraneous-require
     4:24  error  "heimdalljs-logger" is extraneous    node/no-extraneous-require
    24:60  error  "broccoli-middleware" is extraneous  node/no-extraneous-require
  
  lib/tasks/server/middleware/history-support/index.js
    6:30  error  "clean-base-url" is extraneous  node/no-extraneous-require
  
  lib/tasks/server/middleware/proxy-server/index.js
    14:27  error  "http-proxy" is extraneous  node/no-extraneous-require
    28:30  error9m  "morgan" is extraneous      node/no-extraneous-require
  
  lib/tasks/server/middleware/testem-url-rewriter/index.js
    3:30  error  "clean-base-url" is extraneous     node/no-extraneous-require
    4:24  error  "heimdalljs-logger" is extraneous  node/no-extraneous-require
  
  lib/tasks/server/middleware/tests-server/index.js
    3:30  error  "clean-base-url" is extraneous     node/no-extraneous-require
    4:28  error  "exists-sync" is extraneous        node/no-extraneous-require
    7:24  error  "heimdalljs-logger" is extraneous  node/no-extraneous-require
  
  lib/tasks/transforms/amd/index.js
    3:22  error  "broccoli-stew" is extraneous  node/no-extraneous-require
```

I've checked all of these modules. All of them are listed as [dependencies](https://github.com/ember-cli/ember-cli/blob/6917e34a1ba6f06d58c4bb9af329c8d691bf960d/package.json#L88-L89) in the `package.json`. Also there are [lots](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L40) of usages of the same "extraneous" modules across the code base which don't lead to errors.

Any ideas why it can happen? is this a bug?

**Comment by mysticatea** [comment]
Thank you for the report.

This plugin finds `package.json` file based on the location of target JS files.
In that case, [lib/tasks/server/middleware/broccoli-serve-files/package.json](https://github.com/ember-cli/ember-cli/blob/master/lib/tasks/server/middleware/broccoli-serve-files/package.json) was found and it doesn't have dependencies.

**Comment by ro0gr** [comment]
Exactly! Thank you!
